### PR TITLE
Provide more details for serverless deployment + Do not ensure-resource by default to be consistent and document the changes

### PR DESCRIPTION
**Pull request by ahmetoz**
Additionally:
-  Document token cache usage.

### CODE REVIEW

File: deployment-examples/aws/serverless/extension/README.md [hunk 1]
```diff
@@ -1,14 +1,16 @@
...
-      - package.json
-2. Specify `index.handler` as the entry point for the AWS Lambda function.
- > The default value in the console is index.handler which calls exports.handler in index.js
-3. When the extension module is run as AWS Lambda **it will NOT create** the required resources like custom types or commercetools API extension for you. 
-Please follow the [manual resource creation guide](../../../../extension/docs/HowToRun.md#creating-required-resources-manually) instead. 
+1. In order to setup the extension module config, follow our [how to run guide](../../../../extension/docs/HowToRun.md).
+2. Extension module requires `ADYEN_INTEGRATION_CONFIG` as `environment variable`.
+3. Deployments work by uploading an archive containing your function's source code and their dependencies. 
+You could use a deployment package to deploy your function code to Lambda.
+To bundle the extension module, navigate to in `extension` folder and run `npm run zip-lambda-function`.
```
Comment (lojzatran) [inline deployment-examples/aws/serverless/extension/README.md:7]:
```suggestion
To bundle the extension module, navigate into `extension` folder and run `npm run zip-lambda-function`. 
```

File: docs/BEST_PRACTICES.md [hunk 2]
```diff
@@ -1,13 +1,4 @@
...
-<!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->
-**Table of Contents**  
-<!-- *generated with [DocToc](https://github.com/thlorenz/doctoc)* -->
-
-- [Deployment best practices](#deployment-best-practices)
-
-<!-- END doctoc generated TOC please keep comment here to allow auto update -->
-
-#### Deployment best practices
+## Deployment best practices
```
Comment (lojzatran) [inline docs/BEST_PRACTICES.md:1]: What about HMAC? Should we also mention it here?

File: extension/src/ctp.js [hunk 3]
```diff
@@ -12,13 +12,30 @@ const { createQueueMiddleware } = require('@commercetools/sdk-middleware-queue')
 const { createRequestBuilder } = require('@commercetools/api-request-builder')
 const packageJson = require('../package.json')
 
+/**
+The token projectKeyToAuthResultMap is caches the access tokens (based on projectKey) obtained through
```
Comment (lojzatran) [inline extension/src/ctp.js:16]:
```suggestion
The token projectKeyToAuthResultMap caches the access tokens (based on projectKey) obtained through
```

File: extension/src/ctp.js [hunk 4]
```diff
@@ -12,13 +12,30 @@ const { createQueueMiddleware } = require('@commercetools/sdk-middleware-queue')
...
+ the client credentials flow between different clients.
+
+The projectKeyToAuthResultMap is used by middleware to avoid authentication each time when
+ it's need to create an authentication token.
```
Comment (lojzatran) [inline extension/src/ctp.js:20]:
```suggestion
 it's needed to create an authentication token.
```

**Review by lojzatran — CHANGES_REQUESTED** [review]
Not related to this file: I found in `index.googleFunction.js
commercetools-adyen-integration/notification/index.googleFunction.js` that there's no return added in all the places where response is used. Can you add return everywhere? Right now the code might have a weird behaviour.

File: docs/BEST_PRACTICES.md (hunk 2 above)
Comment (ahmetoz) [reply docs/BEST_PRACTICES.md:1]: https://github.com/commercetools/commercetools-adyen-integration/pull/566/commits/90a4bc978833138549ea948eb08122027fa9d843

>

**Comment by ahmetoz** [comment]
> Not related to this PR: I found in `index.googleFunction.js commercetools-adyen-integration/notification/index.googleFunction.js` that there's no return added in all the places where response is used. Can you add return everywhere? Right now the code might have a weird behaviour.

https://github.com/commercetools/commercetools-adyen-integration/pull/566/commits/d6fb74052fc2065239d3f760ef9d30072869976e

**Review by lojzatran — APPROVED** [review]

**Pull request merged**